Reduce IPC queue size from 1024 to 128

cdumez opened the pull request:
#### 4ef72e9956da47077ea460b36d0c593cb32e2f6f
<pre>
Reduce IPC queue size from 1024 to 128
<a href="https://bugs.webkit.org/show_bug.cgi?id=272487">https://bugs.webkit.org/show_bug.cgi?id=272487</a>
<a href="https://rdar.apple.com/125523305">rdar://125523305</a>

Reviewed by Ben Nham.

Reduce IPC queue size from 1024 to 128, to reduce memory usage. Once the queue
size is reached, we wake up the process to give it a chance to process the
queue. Also wake up processes at most once every 20s, instead of 30s.

* Source/WebKit/Platform/IPC/Connection.cpp:

Canonical link: <a href="https://commits.webkit.org/277383@main">https://commits.webkit.org/277383@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e9516282d9f78921339447d0fff29e94abdbb32

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47388 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26569 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50042 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50071 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43436 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32087 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24028 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38581 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47969 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40836 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19901 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5431 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42410 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51947 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22419 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18755 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45881 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23693 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44913 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24483 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6684 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23412 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bnham reviewed (changes requested):
We need to also change this to get the backpressure to kick in faster.

```
    // Change the message queue length for the receive port.
    setMachPortQueueLength(m_receivePort, MACH_PORT_QLIMIT_LARGE);
```

Probably change the second parameter from MACH_PORT_QLIMIT_LARGE to 128 to match.

cdumez committed:
Reduce IPC queue size from 1024 to 128
https://bugs.webkit.org/show_bug.cgi?id=272487
rdar://125523305

Reviewed by Ben Nham.

Reduce IPC queue size from 1024 to 128, to reduce memory usage. Once the queue
size is reached, we wake up the process to give it a chance to process the
queue. Also wake up processes at most once every 20s, instead of 30s.

* Source/WebKit/Platform/IPC/Connection.cpp:

Canonical link: https://commits.webkit.org/277383@main
